docs/source/configuration/traffic-shaping.mdx

```diff
@@ -0,0 +1,38 @@
+---
+title: Traffic shaping
+description: Configuring traffic shaping
+---
+
+import { Link } from "gatsby";
+
+> ⚠️ Apollo Router support for traffic shaping is currently experimental.
+
+The Apollo Router provides experimental support for traffic shaping.
+
+Currently features are limited, but are expected to grow over time:
+
+* [x] **Sub-query deduplication** - Identical in flight sub-queries are compressed into a single request.
+* [ ] Rate limiting
+  * [ ] **Client** - Requests to the Apollo Router are rate limited.
+  * [ ] **Sub-query** - Requests to a sub-service are rate limited.
+* [ ] **Sub-query retry** - Requests to a sub-service are retried if they have failed or timed out.
+
+To configure traffic shaping use the following format:
+
+```yaml
+plugins:
+  experimental.traffic_shaping:
+    all:
+      dedup: true # Enable dedup for all subgraphs.
+    subgraphs: 
+      products:
+        dedup: false # Disable dedup for products.
+```
+
+Note that configuration in the `subgraphs` section will take precedence over that in the `all` section.
+
+## Sub-query deduplication
+
+Deduplication will cause any identical concurrent requests to a subservice to be merged into a single request. This can reduce network bandwidth and CPU at your subgraph.
+
+Note that only in flight requests are deduplicated.
```

Comment:
could you indicate that deduplication only happens on queries, not mutations?

Reply:
Done